### ISSUE TITLE
nullmailer shouldn't install itself only to be removed by postfix in the same run

### ISSUE THREAD

**Issue by jbicha** [debops/ansible-nullmailer#4]
Test case
========
My remote server is a clean Ubuntu 16.04.
Here's an excerpt from `inventory/hosts`
```
[debops_all_hosts]
EXAMPLE.domain.com

[debops_service_postfix]
EXAMPLE.domain.com
```

What happens
============
When I run `$ debops `initially, the nullmailer role installs nullmailer and sets up its configuration.

But a little while later, the postfix role purges nullmailer.

What I expected to happen
========
The nullmailer role detects that this host will have postfix installed and does not install itself.

Maybe you could check if the current host is part of debops_service_postfix?
But some systems don't directly depend on the postfix service but have it as a dependency.
Could you have postfix run before nullmailer?

**Comment by drybjed** [comment]
You can set `nullmailer__enabled: False` in the Ansible inventory for this host, that should tell Ansible to skip installing it. Other than that, there's no reliable and idempotent way for Ansible to detect that you want to install Postfix on that host.

Reordering the `site.yml` playbook so that Postfix role is executed before the common playbook might seem like an easy solution... Of course you probably want to configure the firewall for SMTP access as well so `ferm` would need to be included in the Postfix play for this to happen. And Postfix uses X.509 certificates so `debops.pki` would need to be there somewhere. And `debops.pki` could very much use `debops.dhparam` to have proper Diffie-Hellman parameters to use. And `debops.dhparam` would like to have `debops.atd` handy so that it can schedule DHparam regeneration in the background...

And pretty much you end up with half of the common playbook that needs to be executed before `debops.postfix`.

**Comment by jbicha** [comment]
Yes, I have `nullmailer__enabled: False` set now (I forgot to mention the workaround)
